### ISSUE TITLE
HMCharacteristicPowerState = Mute, when Track==TV

### ISSUE THREAD

**Issue by JotWee** [ebaauw/homebridge-zp#12]
enhancement request:
I would like the power switch to behave like the mute switch, if track is TV

**Comment by NovaGL** [comment]
How would you know if the track is tv? It doesn't show as tv

**Comment by JotWee** [comment]
of course it does

**Comment by ebaauw** [comment]
homebridge-zp shows the track as TV when a PlayBar is playing digital audio input, just as the Sonos app.

In the Sonos app, the Play/Pause (and Next/Previous track) controls are disabled when the PlayBar is playing digital audio input.  Ideally, homebridge-zp would do the same, to provide consistent behaviour, but HomeKit doesn't like characteristics dis- and re-appearing.  It would delete the characteristic from any scenes and rules.

I agree that it makes sense to provide some On/Off functionality when the PlayBar is playing digital input.  Personally, I was thinking about switching (back) to another source, which would effectively mute the PlayBar, but not un-mute it.  See also issue #8 and https://github.com/bencevans/node-sonos/issues/148.

**Comment by ebaauw** [comment]
@JotWee what app is that?

**Comment by JotWee** [comment]
„myHome"
https://appsto.re/de/pmoC_.i

**Comment by ebaauw** [comment]
I figured out how to detect whether the Playbar is actually receiving a signal on the optical input.  The `Power` characteristic now reflects that while the Playbar is on TV input.  Also, homebridge-zp doesn't accept a change to the `Power` characteristic while the Playbar is on TV input.

Technically, I could mute the Playbar (group) when switching off (group) power while it's on TV input.  However, I wouldn't behave like `Mute`:
- The group would go on mute while there's an input signal, but `Power` would remain on;
- You wouldn't be able to unmute the group using `Power` while there is an input signal, as `Power` would still be on;
- The group would go on unmute while there's no input signal, but `Power` would remain off, and there would be no sound coming from the Playbar;
- You wouldn't be able to mute the group using `Power` while there's no input signal, as `Power` would still be off.

I still haven't figured out how to switch the Playbar input (issue #8).

**Comment by ebaauw** [comment]
Apparently, there must be a way to make the Playbar switch back to its previous input.  I would do that when setting `Power` off while on TV input.

I configured my Playbar to switch to TV input automatically, when it detects a signal on the optical input (in the Sonos app select TV Autoplay, under Preferences, Room Settings, TV Input).  It switches back automatically to the previous input after 10 minutes when there's no more signal on the optical input.  I can now see that in the homebridge log:

> [5/2/2017, 11:11:21 AM] [ZP] Living Room Sonos: power (play/pause) changed from true to false
[5/2/2017, 11:21:22 AM] [ZP] Living Room Sonos: current track changed from TV to CITY FM Classic Rock

**Comment by ebaauw** [comment]
Apart from the selected input, the Sonos API also reports whether a signal is detected on the optical input (or on the line-in of a Play:5 or Connect).  Should probably use that to mark player on/off.

**Comment by ebaauw** [comment]
I've added a boolean _TV_ characteristic in v0.3.6 for the PlayBar and (in v0.3.7) PlayBase.  This can be used in HomeKit rules to take action when the TV is turned off.  It's a bit delayed (30s) by design, as sometimes the PlayBar briefly reports no input while the TV is switching audio mode.

Still on my to-do list is a way to select input from HomeKit (see #8), in a way that that also can be done from HomeKit automations.  I probably need the iOS 12.2 _Television_ services for that, see #62.

Closing this one.
